src/server/debug/server/server.go

```diff
@@ -212,7 +212,7 @@ func (s *debugServer) handlePipelineRedirect(
 	collectWorker collectWorkerFunc,
 	redirect redirectFunc,
 ) (retErr error) {
-	prefix := join(pipelinePrefix, pipelineInfo.Pipeline.Name)
+	prefix := join(pipelinePrefix, pipelineInfo.Pipeline.Project.Name, pipelineInfo.Pipeline.Name)
```

Comment:
This pattern in general worries me; if somehow things are in a bad state and project is nil, then we panic.  I would prefer using the accessors throughout: pipelineInfo.GetPipeline().GetProject().GetName().  These will not panic when nil, even when chained.

Reply:
I like your fixup here.